### PR TITLE
Add auth, account, login realm and sign up sdk and their mocks

### PR DESCRIPTION
**Pull request by MuweiHe**
`Auth`, `login realm` are required for cli to handle login process. `sign up` is required before cdmum team works on making captcha work with cli. `Account` is added as I want to use one single client instead of both v1-private and v1-public in verifying environment when login in.

reorganize the package

tested this branch with cli

### CODE REVIEW

File: ccloud/mock/Account.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
+// Code generated by mocker. DO NOT EDIT.
```
Comment (brianstrauch) [inline ccloud/mock/Account.go:1]: The name of this file should be lowercase.

File: ccloud/go.mod [hunk 2]
```diff
@@ -1,4 +1,4 @@
-module github.com/confluentinc/ccloud-sdk-go-v1-public
+module github.com/confluentinc/ccloud-sdk-go-v1-public/ccloud
```
Comment (brianstrauch) [inline ccloud/go.mod:1]: Are you planning on putting another sub-package in this repo? What's the reason for this?

**Review by brianstrauch — CHANGES_REQUESTED** [review]

File: ccloud/go.mod (hunk 2 above)
Comment (MuweiHe) [reply ccloud/go.mod:1]: Feels more organized this way as I was planning on adding v1 mds sdk. The plan is to only release v1 on-prem mds and put cloud role api in v1-public.

Comment (brianstrauch) [reply ccloud/go.mod:1]: It still feels weird cramming multiple repos in here, how do you feel about creating a mds-sdk-go-public for the MDS cloud SDK?

Comment (MuweiHe) [reply ccloud/go.mod]: ok let's do that!

**Review by brianstrauch — APPROVED** [review]

**Pull request merged**